mbus-api/src/mbus_nats.rs

```diff
@@ -0,0 +1,330 @@
+            max_ack_pending: 1, /* NOTE: when this value is 1, we receive and ack each message
+                                 * before moving on to the next ordered message */
+            ..Default::default()
+        };
+
+        loop {
+            let err = match stream
+                .get_or_create_consumer(CONSUMER_NAME, consumer_config.clone())
+                .await
+            {
+                Ok(consumer) => match consumer.messages().await {
+                    Ok(messages) => {
+                        tracing::info!(
+                            "Getting/creating consumer for stats '{}' successful",
+                            CONSUMER_NAME
+                        );
+                        return Ok(messages);
+                    }
+                    Err(error) => error,
+                },
+                Err(error) => error,
+            };
+
+            if tries == options.clone().max_retries {
+                return Err(JetstreamError::ConsumerError {
+                    consumer: CONSUMER_NAME.to_string(),
+                    error: err.to_string(),
+                });
+            }
+            if log_error {
+                tracing::warn!(
+                    "Nats error while getting consumer '{}' messages: {:?}. Retrying...",
+                    CONSUMER_NAME,
+                    err
+                );
+                log_error = false;
+            }
+            backoff_with_options(&mut tries, options.clone()).await;
+        }
+    }
+
+    /// Creates a stream if not exists on message bus. Returns a handle to the stream
+    pub async fn get_or_create_stream(
+        &self,
+        retry_options: Option<BackoffOptions>,
+    ) -> BusResult<Stream> {
+        tracing::info!("Getting/creating stream '{}'", STREAM_NAME);
+        let options = retry_options.unwrap_or(BackoffOptions::new());
+        let mut tries = 0;
+        let mut log_error = true;
+        let stream_config = async_nats::jetstream::stream::Config {
+            name: STREAM_NAME.to_string(),
+            subjects: vec![SUBJECTS.into()],
+            duplicate_window: DUPLICATE_WINDOW,
+            max_bytes: STREAM_SIZE,
+            storage: async_nats::jetstream::stream::StorageType::Memory, /* The type of storage
+                                                                          * backend, `File`
+                                                                          * (default) */
+            num_replicas: NUM_STREAM_REPLICAS,
+            ..async_nats::jetstream::stream::Config::default()
+        };
+
+        loop {
+            let err = match self
+                .jetstream
+                .get_or_create_stream(stream_config.clone())
+                .await
+            {
+                Ok(stream) => {
+                    tracing::info!("Getting/creating stream '{}' successful", STREAM_NAME);
+                    return Ok(stream);
+                }
+                Err(error) => error,
```

Comment:
Return  specific error using Snafu, this way it will be easy for debugging in case of errors.

Reply:
The error has the error message from Nats which can be used for debugging. Should I still separate the errors?